lib/sda/host_msg.go

```diff
@@ -56,6 +63,8 @@ func (t *Token) Id() uuid.UUID {
 
 // Return a new Token contianing a reference to the given TreeNode
 func (t *Token) ChangeTreeNodeID(newid uuid.UUID) *Token {
+	tokenMutex.Lock()
+	defer tokenMutex.Unlock()
```

Comment:
can you explain why we need a mutex here ?
It seems to me that the only reason we need the token is because of this `ChangeTreeNodeID` ?
If so, we should find another way to solve the data race problem because token.Id() is **often** used and since Token *should* be immutable (even if we did not enforce it *yet*), there *should* be no need of mutex.